flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/functions/ScalarFunctions.scala

```diff
@@ -196,9 +196,35 @@ object ScalarFunctions {
     new String(data)
   }
 
+  /**
+    * Returns the numeric value of the leftmost character of the string str.
+    * Returns 0 if str is the empty string. Returns NULL if str is NULL.
+    */
+  def ascii(str: String): String = {
+    if (str == null) {
+      null
+    } else if (str.equals("")) {
+      ""
+    } else {
+      str.charAt(0).toByte.toString
+    }
+  }
+
   /**
     * Returns the base string decoded with base64.
     */
   def fromBase64(str: String): String = new String(Base64.decodeBase64(str))
 
+  /**
+    * Returns string contains a character which converts from a ASCII integer.
+    * If the ASCII less then 0 or greater than 255, return null.
+    */
+  def chr(ascii: Integer): String = {
```

Comment:
For the function signature, I think we should support other types, say Byte/Short/Long. Exceptions will be thrown if we write `chr(CAST(1 AS SMALLINT))`.

Reply:
hi @hequn8128 others has accepted and fixed except this. can you give a hint? you mean:

* one function and the parameter support multiple types? if support I can you Either[Type1, Type2], but more than two, how to handle this? I am not very familiar with Scala.

* multiple functions, one support a single type?

* assume supported these types, the `OperandTypes.NUMERIC` can represent them?

```scala
val CHR = new SqlFunction(
    "CHR",
    SqlKind.OTHER_FUNCTION,
    ReturnTypes.cascade(ReturnTypes.explicit(SqlTypeName.VARCHAR), SqlTypeTransforms.TO_NULLABLE),
    null,
    OperandTypes.NUMERIC,
    SqlFunctionCategory.STRING
  )
```

Reply:
@yanghua Sorry that haven't make it clear.
Yes, what i mean is add multi-functions. You probably have to add the following changes:
1. add different functions in `ScalarFunctions.scala`.
2. add other `addSqlFunctionMethod` in `FunctionGenerator`. 
```
  addSqlFunctionMethod(
    CHR,
    Seq(LONG_TYPE_INFO),
    STRING_TYPE_INFO,
    BuiltInMethods.CHR
  )
```

Reply:
@hequn8128  maybe we could not add more functions to support `byte/short/int/long` type, I just tried change all the specific `Integer` to `Long` about this function, then tested : 

```
testAllApis(
      'f36.chr(),
      "f36.chr()",
      "CHR(f36)",
      "A")

    testAllApis(
      'f36.chr(),
      "f36.chr()",
      "CHR(CAST(f36 AS SMALLINT))",
      "A")

    testAllApis(
      'f36.chr(),
      "f36.chr()",
      "CHR(CAST(f36 AS TINYINT))",
      "A")

    testAllApis(
      'f36.chr(),
      "f36.chr()",
      "CHR(CAST(f36 AS BIGINT))",
      "A")
```

all test passed

Reply:
It's weird. I checkout your latest branch and find the exception still will be thrown.

Reply:
yes, I said I changed `Integer` to `Long` in my local env, the test is OK. I did not update this PR. I just discuss with you whether solution is OK or not. The code you tested will failed because of `SMALLINT ` or `TINYINT `?

Reply:
It's true. The code before latest commit, will also fail for `TINYINT `, `SMALLINT `. The latest commit can pass these test cases.

Reply:
Cool, it works now.